RFC-004-Bitcoin.md

```diff
@@ -0,0 +1,54 @@
+# Bitcoin Definitions
+
+As per [Bitcoin Core](https://github.com/bitcoin/bitcoin/) concensus.
+
+Introduced for RFC-003 (should RFC-003 be part of the title?)
+
+## Ledger definition
+
+Taken from registry
+
+| Name     | Description            | Reference |  `value`   | `parameters` key | `parameters` value |`parameters` description         |
+|:---      |:---                    |:---       |:---        |:---              |:---                |:---                             |
+| Bitcoin  | Bitcoin-core network   | TBD       | `bitcoin`  | `network`        |                    | The network on which to operate |
+|          |                        |           |            |                  | `regtest`          | Bitcoin-core regtest            |
+|          |                        |           |            |                  | `testnet`          | Bitcoin testnet                 |
+|          |                        |           |            |                  | `mainnet`          | Bitcoin mainnet                 |
+
+## Asset definition
+
+Taken from registry
+
+| Name           | Description                   | Reference | `value`   | `parameters` key | `parameters` value type | `parameters` description |
+|:---            |:----                          |:---       |:---       |:---              |:---                     |:---                      |
+| Bitcoin        | Native Bitcoin network asset  | TBD       | `bitcoin` | `quantity`       | integer in Json string  | Amount in satoshi        |
+
+## Identities
+
+Taken from registry
+
+| Ledger   | Identity Name | JSON Encoding            | Reference | Description                                                                       |
+|:----     |:-------      |:-------------            |:--------- | --------------------------------------------------------------------------------- |
+| Bitcoin  | `pubkeyhash`  | `hex-encoded-bytes (20)` | TBD       | The result of applying SHA256 and then RIPEMD160 to a user's SECP256k1 public key |
+
+## RFC-003
+
+Mention that it should be p2wsh.
```

Comment:
Eventually we might support other addresses (e.g. if we create a connector to bitcoin cash, etc) but for now I would say we only support p2wsh.

Reply:
Bitcoin cash is a different ledger IMO and hence a different RFC.

Reply:
true, but then we will have support for both address types.

Reply:
The implementation will hopefully be the same or at least share many things. The RFC would just be the specification of "this is how you get Bitcoin Cash to work with COMIT".

Because it is a fork of Bitcoin, this spec could just say:

- this is how you identify the BCH ledger
- we add support for these address types under these conditions
- whatever else is needed

It doesn't mean that things will be totally separate :)

Reply:
I don't think Bitcoin Cash (or OL), matters here.
If we only support P2WSH then we stop users to do a swap with a number of wallets.
I think considering P2SH or P2SH(P2WSH) would allow user to use *any* wallet.

Reply:
My thought was actually that we only support P2WSH  for now and once we support BCash (which will be an own RFC) we can revisit this document and add support for other addresses as the implementation will be the same.